demos/CMakeLists.txt

```diff
@@ -35,6 +35,38 @@ function(check_aws_credentials demo_name)
     endif()
 endfunction()
 
+# This function will add a demo target to the list of default targets
+# if pre-signed GET and PUT URLs are defined.
+function(check_presigned_urls demo_name)
+    if(S3_PRESIGNED_GET_URL)
+        target_compile_definitions(
+            ${demo_name} PRIVATE
+                S3_PRESIGNED_GET_URL="${S3_PRESIGNED_GET_URL}"
+        )
+        set(CMAKE_REQUIRED_DEFINITIONS -DS3_PRESIGNED_GET_URL)
+    endif()
+
+    if(S3_PRESIGNED_PUT_URL)
+        target_compile_definitions(
+            ${demo_name} PRIVATE
+                S3_PRESIGNED_PUT_URL="${S3_PRESIGNED_PUT_URL}"
+        )
+        set(CMAKE_REQUIRED_DEFINITIONS -DS3_PRESIGNED_GET_URL -DS3_PRESIGNED_PUT_URL)
+    endif()
+
+    set(FILES_TO_CHECK "demo_config.h")
+    list(APPEND CMAKE_REQUIRED_INCLUDES "${CMAKE_CURRENT_LIST_DIR};${LOGGING_INCLUDE_DIRS}")
+    unset(HAVE_S3_PRESIGNED_GET_URL CACHE)
+    unset(HAVE_S3_PRESIGNED_PUT_URL CACHE)
+    check_symbol_exists(S3_PRESIGNED_GET_URL ${FILES_TO_CHECK} HAVE_S3_PRESIGNED_GET_URL)
+    check_symbol_exists(S3_PRESIGNED_PUT_URL ${FILES_TO_CHECK} HAVE_S3_PRESIGNED_PUT_URL)
+
+    if(NOT(HAVE_S3_PRESIGNED_GET_URL))
+        message("To run ${demo_name}, define S3_PRESIGNED_GET_URL (and S3_PRESIGNED_PUT_URL if applicable) in ${demo_name}/demo_config.h.")
```

Comment:
Seems that `HAVE_S3_PRESIGNED_PUT_URL` is never utilized. Ideally, we would want to check `${demo_name}` and check the required symbol based on it.